### PR TITLE
updates to clustering

### PR DESCRIPTION
**Pull request by jelmerborst**
- Bunch of updates to clustering, to make it faster and make outlier detection optional. Also plotting is better.
- Added 1 for min_estimates in some pool functions in models.py.

### CODE REVIEW

File: src/hsmm_mvpy/models.py [hunk 1]
```diff
@@ -236,12 +237,12 @@ def fit_single(self, n_events=None, magnitudes=None, parameters=None, parameters
 
                     estimates = pool.starmap(self.EM, 
                         zip(itertools.repeat(n_events), magnitudes, parameters, itertools.repeat(maximization),
-                        itertools.repeat(magnitudes_to_fix),itertools.repeat(parameters_to_fix), itertools.repeat(max_iteration), itertools.repeat(tolerance)))   
+                        itertools.repeat(magnitudes_to_fix),itertools.repeat(parameters_to_fix), itertools.repeat(max_iteration), itertools.repeat(tolerance),itertools.repeat(1)))
```
Comment (GWeindel) [inline src/hsmm_mvpy/models.py:240]: This can also be removed so that it always takes the default value

File: src/hsmm_mvpy/models.py [hunk 2]
```diff
@@ -236,12 +237,12 @@ def fit_single(self, n_events=None, magnitudes=None, parameters=None, parameters
...
                     estimates = pool.starmap(self.EM, 
                         zip(itertools.repeat(n_events), magnitudes, parameters, itertools.repeat(maximization),
-                        itertools.repeat(magnitudes_to_fix),itertools.repeat(parameters_to_fix), itertools.repeat(max_iteration), itertools.repeat(tolerance)))   
+                        itertools.repeat(magnitudes_to_fix),itertools.repeat(parameters_to_fix), itertools.repeat(max_iteration), itertools.repeat(tolerance),itertools.repeat(1)))   
             else:#avoids problems if called in an already parallel function
                 estimates = []
                 for pars, mags in zip(parameters, magnitudes):
                     estimates.append(self.EM(n_events, mags, pars, maximization,\
-                    magnitudes_to_fix, parameters_to_fix, max_iteration, tolerance))
+                    magnitudes_to_fix, parameters_to_fix, max_iteration, tolerance, min_iteration=1))
```
Comment (GWeindel) [inline src/hsmm_mvpy/models.py:245]: Same here

**Review by GWeindel — COMMENTED** [review]
So I trust you on the module you wrote, I just have minor comments on the specification of the minimum_iteration in models.py, it's not necessary (I think) to declare them explicitly. removing them would allow future default value to be automatically applied to the functions I commented

**Comment by jelmerborst** [comment]
fixed, can be merged

**Review by GWeindel — APPROVED** [review]

**Pull request merged**